silkrpc/core/evm_trace.cpp

```diff
@@ -953,11 +976,57 @@ void IntraBlockStateTracer::on_reward_granted(const silkworm::CallResult& result
 }
 
 template<typename WorldState, typename VM>
-asio::awaitable<std::vector<TraceCallResult>> TraceCallExecutor<WorldState, VM>::execute(const silkworm::Block& block) {
+asio::awaitable<std::vector<Trace>> TraceCallExecutor<WorldState, VM>::trace_block(const silkworm::BlockWithHash& block_with_hash) {
+    std::vector<Trace> traces;
+
+    auto trace_call_results = co_await trace_blockTransactions(block_with_hash.block, {false, true, false});
+    for (std::uint64_t pos = 0; pos < trace_call_results.size(); pos++) {
+        silkrpc::Transaction transaction{block_with_hash.block.transactions[pos]};
+        if (!transaction.from) {
+            transaction.recover_sender();
+        }
+        auto hash = hash_of_transaction(transaction);
+        auto tnx_hash = silkworm::to_bytes32({hash.bytes, silkworm::kHashLength});
+
+        auto& trace_call_result = trace_call_results.at(pos);
+        auto& call_traces = trace_call_result.traces.trace;
```

Comment:
`trace_call_result` and `call_traces` should be `const`

Reply:
Of course